core/src/main/java/com/linecorp/armeria/common/stream/AsyncMapStreamMessage.java

```diff
@@ -90,32 +95,29 @@ public void abort(Throwable cause) {
         private final Subscriber<? super U> downstream;
         private final Function<T, CompletableFuture<U>> function;
         private final EventExecutor executor;
+        private final int maxConcurrency;
 
         @Nullable
         private volatile Subscription upstream;
         private volatile boolean canceled;
 
         private long requestedByDownstream;
-        private State state;
-
-        enum State {
-            WAITING,
-            REQUESTED,
-            PROCESSING,
-            COMPLETING
-        }
+        private int requestedFromUpstream;
```

Comment:
nit; wondering if `requestedToUpstream` is more representative of what this variable is tracking

```suggestion
        private int requestedToUpstream;
```

Reply:
I'm a little bit conflicted on this. I think both are valid (English is weird). I'll leave it up to you maintainers 😄